dimagi/commcare-hq
# Keep case search query inputs out of address bar

## Product Description
Case search "inputs" are no longer in the web address bar, so if any projects copy + paste links directly to case search results this will no longer be possible.

## Technical Summary
[USH-3834](https://dimagi-dev.atlassian.net/browse/USH-3834)
Rather than storing search inputs in the url, this change means that query inputs are now stored in and retrieved from `Utils.queryInputs`, under their appropriate `queryKey` to account for times when more than one set of queryData may be present. The same data was previously already stored as `stickyQueryInputs`, so there is no added security concern. This change keeps `stickyQueryInputs` functionality with minor refactoring, as it represents a different state of inputs -- one that is used to repopulate previously entered values, rather than used to construct a request to Formplayer.

## Feature Flag
SYNC_SEARCH_CASE_CLAIM

## Safety Assurance

### Safety story
Tested locally with different case search workflows (split screen & regular) and found basic case search works as expected and keeps query inputs out of the browser URL. Tested on staging with various parent > child select and multi-select workflows, which surfaced issues with the original approach that have now been addressed.

### Automated test coverage
Touched by mocha test "should navigate through case search" showing that the navigation url is ultimately set correctly.

### QA Plan
This should get QA to ensure compatibility with various case search workflows and split screen case search with or without dynamic search.

### Rollback instructions




### Labels & Review


[USH-3834]: https://dimagi-dev.atlassian.net/browse/USH-3834?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
